[docs] add intro content from training module

angelinalg opened the pull request:
## Why are these changes needed?
The docs can use some high level introductory material. The training modules have very nice content to leverage! I included some features of Ray and a description of the libraries.

This is a Work In Progress [WIP] to get some early feedback before fleshing out more.
cc: @emmyscode

pcmoritz commented:
The content looks great, have you considered putting it into https://docs.ray.io/en/latest/index.html which is the canonical landing page for the Ray documentation? And then link it from "Getting started" with a line like "If you want to learn more about what Ray is and what you can do with it, see [here]" at the top? That would make sure that content is not duplicated and the "Getting started" guide keeps getting to the point quickly, yet giving people that don't know about Ray the opportunity to learn :)

angelinalg commented:
You are so right, @pcmoritz. I broke the cardinal rule of mixing content styles. I will rework!

angelinalg commented:
Summary of changes:
1. Moved overview content from training to the home page.
2. Moved the "Why Ray?" to the intro paragraph, and "What is Ray? before "What can you do" to improve the flow of content. 
3. Removed the "What documentation resource is right for you?" because with the recent IA changes, the navigation is intuitive.
4. Moved the logo further down so that user doesn't need to scroll as much to get to content.
5. Tightened up the text for the cards in "What can you do".
Before:
<img width="791" alt="Screenshot 2023-02-07 at 11 39 24 AM" src="https://user-images.githubusercontent.com/122562471/217349140-6a10f77d-ad5b-4e1e-8ba2-4bf8c47ee4c8.png">
After:
<img width="786" alt="Screenshot 2023-02-07 at 11 39 10 AM" src="https://user-images.githubusercontent.com/122562471/217349170-0a442912-2585-4603-9416-9c6c8b3b6e1e.png">

richardliaw commented:
this is... beautiful....

maxpumperla commented on doc/source/ray-overview/what-is-ray.md:
```diff
@@ -0,0 +1,77 @@
+# What is Ray
+
+<strong><a href="https://www.ray.io/" target="_blank">Ray</a></strong> is an open-source unified compute framework that makes it easy to scale AI and Python workloads. This guide gives a quick tour of Ray's features.
```
@angelinalg I think you left this document in by accident, correct? All changes should be in  `index.md`.

maxpumperla replied:
once you remove this doc, the CI build should also pass.

angelinalg replied:
Thanks for catching that, @maxpumperla! I'll take care of it.

maxpumperla commented on doc/source/index.md:
````diff
@@ -2,200 +2,226 @@
+[latest version of our contributing guide](https://docs.ray.io/en/latest/ray-contribute/getting-involved.html)
+to read about the contribution process and see what you can work on.
+
+```{image} https://github.com/ray-project/ray/raw/master/doc/source/images/ray_header_logo.png
````
you moved the header and all logos to the bottom (see rendered page). I don't think that's what we want. The badges we can discuss, but a prominent logo should still be featured up top.

angelinalg replied:
Let me share why I removed it from the top. I thought that it was generally a bad user experience to have to scroll through graphics to get to content. The logo is also in the upper left hand corner, so the larger logo at the top is redundant.

I referenced other docs home pages to check my bias:

- https://docs.databricks.com/
- https://modal.com/docs/guide (their docs have been coming up in conversations recently)
- https://stripe.com/docs
- https://www.twilio.com/docs
- https://platform.openai.com/docs/introduction
- https://docs.grid.ai/

Those examples all have the logo in the upper left corner. WDYT, @maxpumperla? Let me know if you think my examples are too skewed.

simran-2797 replied:
Hey! I like it! I do agree that we do not need the logo on top of the page since we already have the Ray logo on the horizontal navigation bar. We also have the text "Ray 3.0.0" on the left vertical navigation, so putting the Ray logo again in the page might be redundant.

angelinalg replied:
Yes, I had the same feeling of splitting out the ML Workloads into a separate page. Thanks, @simran-2797! I'll file an issue to "freeze" the top nav.

maxpumperla replied:
@angelinalg ok, understood. removing the logo is fine. I just thought it looked silly to put it at the bottom - better to remove altogether.

angelinalg replied:
I removed the logo. Thanks, @maxpumperla.

maxpumperla commented on doc/source/index.md:
```diff
@@ -2,200 +2,226 @@
 
-<img src="images/user_guide.svg" alt="user_guide" height="40px">
-^^^^^^^^^^^
+### Multi-model composition for model serving
```
I think the image for this one is really hard to see. Maybe we can just have parts of it, or mention that you can "click to enlarge" or so.

emmyscode replied:
This diagram has a lot going on. A new version is in the works for upcoming serving tutorials, but in the meantime, I can forward along some alternatives.

angelinalg replied:
I'll add a note to "Click image to enlarge." in the caption. @emmyscode, I'd love to update it when the new version is available. Thanks!

maxpumperla commented on doc/source/index.md:
```diff
@@ -2,200 +2,226 @@
-Our user guides provide you with in-depth information about how to use Ray's libraries and tooling.
-You will learn about the key concepts and features of Ray and how to use them in practice.
-+++
+[Ray Serve](https://docs.ray.io/en/latest/serve/index.html) supports complex [model deployment patterns](https://www.youtube.com/watch?v=mM4hJLelzSw) requiring the orchestration of multiple Ray actors, where different actors provide inference for different models. Serve handles both batch and online inference and can scale to thousands of models in production.
```
Probably not for this PR, but I think we're already using this plugin https://github.com/sphinx-contrib/youtube to embed yt videos directly. Having more multi-modal content is important, and I think going through our yt channel and take inventory in terms of what we can cross-link on our docs could be extremely valuable.

angelinalg replied:
Good to know. I agree that we could do more cross-linking in general. Thanks!

maxpumperla reviewed (changes requested):
@angelinalg really like this! just left a couple of comments to address.

pcmoritz commented:
This looks great to me now, thanks a lot for doing this -- let me know when it is ready to give a little more detailed feedback. I'd consider putting the badges back on top so people can find our slack etc. quickly but up to you.

angelinalg commented:
Addressed comments from @maxpumperla, @Wendi-anyscale, and @simran-2797, and input from @richardliaw.
1. Split home page into a landing page (index.md) and an overview page (ray-overview/index.md), and added "Overview" to the TOC under "Ray".
2. Moved "Common ML workloads" into a separate page and linked to it from the "Overview" page. (Having a problem making "Common ML workloads" a child page of "Overview". Any tips would be appreciated!)
3. Replaced intro paragraph of "Overview" page (which was the intro paragraph of the original home page), by an intro that ML practitioners can relate more to.

Trying to get this signed off today to be included in 2.3 so please let me know if you have any blocking comments! Thanks!

angelinalg commented:
@maxpumperla, I had trouble making "Common ML workloads" a child page of "Overview". After running some experiments, it appears that the problem might be that "Overview" is an MD file. I didn't know if you have a preferred MD>RST convertor, so I left it on the top-level. If you can easily convert it, or somehow figure out how an MD file can take a child page, would you mind doing it for me? Thanks!

angelinalg commented:
@cadedaniel, please include this in 2.3 after it gets approved. TY!

maxpumperla commented:
@angelinalg your problems, I believe come from the fact that you moved contents of a file from the base folder to `ray-overview`. That means all those relative references, like `data/datasets.rst` can't be found anymore (obv. just use `../data/datasets.rst` etc.).

Personally, I'm a proponent of myst markdown over rst, as the latter is just not a great format to read or write. If you have to know, rst to myst conversion is best done with [this python lib](https://pypi.org/project/rst-to-myst/), and the converse you could do with [pandoc](https://gist.github.com/zaiste/77a946bbba73f5c4d33f3106a494e6cd). Just note that that won't solve your problems, you have to apply the fixes as indicated above.

Makes sense? We can't merge this before all doc tests pass.

angelinalg commented:
@pcmoritz, the badges are back up on top of the home page. Let me know if you have any more blocking issues! Thx!

Wendi-anyscale commented:
Love the changes! These definitely help fix the issues we talked about.
Would it be great to use the same format for these 2 paragraphs and call out the persona in the front?
e.g.

- For Data scientist
- For Distributed system engineers…
<img width="674" alt="Screenshot 2023-02-14 at 12 00 23 PM" src="https://user-images.githubusercontent.com/97252176/218873658-5dbb9fe0-758c-4112-84f7-6ad1439e0257.png">
 and machine learning practitioners….

richardliaw commented on doc/source/_toc.yml:
```diff
@@ -2,8 +2,13 @@ format: jb-book
   - caption: Ray
-    chapters:
+    chapters: 
       - file: ray-overview/index
+        title: "Overview"
+        sections:
+          - file: ray-overview/workloads
+            title: "Common ML workloads"
```
<img width="259" alt="image" src="https://user-images.githubusercontent.com/4529381/218892393-dc743f29-14f7-4f0a-a2af-fe204698a9ce.png">
this ends up rendering a bit oddly (it wasn't clear that the header was clickable)

angelinalg replied:
You mean the carrot didn't show up?

richardliaw replied:
no, i mean that it wasnt clear to me that you could actually click the overview page

angelinalg replied:
Ah. Yes, this is a source of debate, whether we should put the "Common ML workloads" page at the top level. Max was trying to keep the top level short because it was really bloated. Are you suggesting that I pull it up to the top level?

angelinalg replied:
Maybe what's not clear is that the `Overview` page links to the `Common ML workloads` page: 
<img width="723" alt="Screenshot 2023-02-14 at 5 36 31 PM" src="https://user-images.githubusercontent.com/122562471/218905207-5125b683-7679-46ec-b2df-f94b226f2780.png">

angelinalg replied:
I pulled the Workloads page to the top level in the side nav.
<img width="260" alt="Screenshot 2023-02-14 at 6 38 56 PM" src="https://user-images.githubusercontent.com/122562471/218913322-05529083-f733-43f3-b70e-74c43bf65703.png">

richardliaw commented:
Seems like there's some conflicts?

pcmoritz reviewed (approved):
This looks great, thanks for doing it!
